src/containers/pages/FocusInvestigation/map/active/index.tsx

```diff
@@ -153,319 +143,272 @@ export const defaultMapSingleFIProps: MapSingleFIProps = {
   supersetService: supersetFetch,
 };
 
-/** Map View for Single Active Focus Investigation */
-class SingleActiveFIMap extends React.Component<
-  RouteComponentProps<RouteParams> & MapSingleFIProps,
-  FlexObject
-> {
-  public static defaultProps = defaultMapSingleFIProps;
-  constructor(props: RouteComponentProps<RouteParams> & MapSingleFIProps) {
-    super(props);
+interface StatusBadgeProps {
+  plan: Plan;
+}
+
+const StatusBadge = (props: StatusBadgeProps) => {
+  const { plan } = props;
+  const color =
+    (plan && plan.plan_status === PlanStatus.ACTIVE) ||
+    (plan && plan.plan_status === PlanStatus.DRAFT)
+      ? 'warning'
+      : 'success';
+
+  return (
+    <Badge color={color} pill={true}>
+      {plan && plan.plan_status}
+    </Badge>
+  );
+};
+
+interface MarkCompleteLinkProps {
+  plan: Plan;
+}
+
+const MarkCompleteLink = (props: MarkCompleteLinkProps) => {
+  const { plan } = props;
+
+  if (plan && plan.plan_status === PlanStatus.ACTIVE) {
+    return (
+      <Link className="btn btn-primary" to={`${PLAN_COMPLETION_URL}/${plan.id}`} color="primary">
+        {MARK_AS_COMPLETE}
+      </Link>
+    );
+  }
+
+  return null;
+};
+
+const getDetailViewPlanInvestigationContainer = (plan: Plan): React.ReactElement[] => {
+  const detailViewPlanInvestigationContainer: React.ReactElement[] = [];
+  /** Array that holds keys of a plan object
+   * Will be used to check plan_effective_period_end or plan_effective_period_start to build the detailview
+   */
+  const planKeysArray: string[] = Object.keys(plan);
+
+  /** alias enables assigning keys dynamically used to populate the detailview  */
+  const alias = {
+    plan_effective_period_end: END_DATE,
+    plan_effective_period_start: START_DATE,
+  };
+
+  if (plan.plan_fi_reason === ROUTINE) {
+    planKeysArray.forEach((column: string) => {
+      if (column === 'plan_effective_period_end' || column === 'plan_effective_period_start') {
+        detailViewPlanInvestigationContainer.push(
+          <span key={column} className="detailview">
+            <b>{alias[column]}:</b> &nbsp;
+            {plan && plan[column]}
+          </span>
+        );
+      }
+    });
+  } else {
+    detailViewPlanInvestigationContainer.push(
+      <span key={plan && plan.plan_id}>
+        <b>{CASE_CLASSIFICATION_LABEL}:</b>&nbsp;
+        {plan && plan.plan_intervention_type ? plan.plan_intervention_type : null}
+      </span>
+    );
   }
 
-  public componentDidMount() {
-    if (!this.props.plan) {
+  return detailViewPlanInvestigationContainer;
+};
+
+/** Map View for Single Active Focus Investigation */
+const SingleActiveFIMap = (props: MapSingleFIProps & RouteComponentProps<RouteParams>) => {
```

Comment:
I am curious about your thoughts on this component as regards the [Open/Closed principle](https://stackify.com/solid-design-open-closed-principle/).  I am specifically wondering about its usage of sub components, and functions such as `fetchData` and whether it should take them in as props.  What do you think? (This is not a directive but a question)

Reply:
@moshthepitt Mmmh...What do you think might change in future that would help if we passed it as a prop? Also, does it mean we extend the functionality of `fetchData` without affecting the render method of the component?

Reply:
@kelvin-muchiri that's a good question.  I'll leave it to you to decide if we need to do anythin regarding that.

Reply:
@moshthepitt I haven't thought about it yet. Maybe we can proceed at it is for now.

Reply:
Cool @kelvin-muchiri

Reply:
@moshthepitt Is the coveralls build not completing because the base branch is not master?

Reply:
Yeah we need @p-netm to update tiny-backend with master, and then you can do that for this branch as well.  But I am fine merging this before then.